src/Persistence/Mapping/Driver/ColocatedMappingDriver.php

```diff
@@ -132,35 +123,22 @@ public function getAllClassNames(): array
             return $this->classNames;
         }
 
-        if ($this->paths === []) {
+        if ($this->paths === [] && ! isset($this->filePaths)) {
             throw MappingException::pathRequiredForDriver(static::class);
         }
 
-        /** @var AppendIterator<array-key,SplFileInfo,Iterator<array-key,SplFileInfo>> $filesIterator */
-        $filesIterator = new AppendIterator();
-
-        foreach ($this->paths as $path) {
-            if (! is_dir($path)) {
-                throw MappingException::fileMappingDriversRequireConfiguredDirectoryPath($path);
-            }
+        $dirFilesIterator = new DirectoryFilesIterator($this->paths, $this->fileExtension);
 
-            /** @var Iterator<array-key,SplFileInfo> $iterator */
-            $iterator = new RegexIterator(
-                new RecursiveIteratorIterator(
-                    new RecursiveDirectoryIterator($path, FilesystemIterator::SKIP_DOTS),
-                    RecursiveIteratorIterator::LEAVES_ONLY,
-                ),
-                sprintf('/%s$/', preg_quote($this->fileExtension, '/')),
-                RegexIterator::MATCH,
-            );
-
-            $filesIterator->append($iterator);
-        }
+        /** @var iterable<string> $filePathsIterator */
+        $filePathsIterator = $this->concatIterables(
+            $this->filePaths ?? [],
+            new FilePathNameIterator($dirFilesIterator),
+        );
```

Comment:
The `AppendIterator` would be perfect to replace `concatIterables` and avoid this private method to the class that uses the trait.
```suggestion
        $filePathsIterator = new \AppendIterator();
        $filePathsIterator->append(new ArrayIterator($this->filePaths ?? []));
        $filePathsIterator->append(new FilePathNameIterator($dirFilesIterator));
```

Reply:
I also considered doing it this way, but the problem is that `ArrayIterator` doesn't accept a generic `iterable` (nor `Traversable`). Although it works well with `array|ArrayObject`, that's the only input type it supports. It won't work with `$filePaths` being some `Traversable` instead.